src/commands/client/upsert/index.ts

```diff
@@ -2,14 +2,36 @@ import { Command } from "@commands/command";
 
 type NoInfer<T> = T extends infer U ? U : never;
 
-type UpsertCommandPayload<TMetadata> = {
+type VectorPayload<TMetadata> = {
   id: number | string;
   vector: number[];
   metadata?: NoInfer<TMetadata>;
 };
+type DataPayload<TMetadata> = {
+  id: number | string;
+  data: string;
+  metadata?: NoInfer<TMetadata>;
+};
+
+type PayloadArray<TMetadata> = VectorPayload<TMetadata>[] | DataPayload<TMetadata>[];
 
 export class UpsertCommand<TMetadata> extends Command<string> {
-  constructor(payload: UpsertCommandPayload<TMetadata> | UpsertCommandPayload<TMetadata>[]) {
-    super(payload, "upsert");
+  constructor(
+    payload: VectorPayload<TMetadata> | DataPayload<TMetadata> | PayloadArray<TMetadata>
```

Comment:
do we also hint that these can be arrays?
previously it was used

```
UpsertCommandPayload<TMetadata> | UpsertCommandPayload<TMetadata>[]
```

Reply:
Yeah. There are 3 different variants

- VectorPayload<TMetadata>[] | DataPayload<TMetadata>[];
- VectorPayload<TMetadata>
- DataPayload<TMetadata>

And, old tests are still working fine. Everything is backwards compatible